crates/ruff_python_formatter/src/statement/stmt_class_def.rs

```diff
@@ -117,14 +117,30 @@ impl FormatNodeRule<StmtClassDef> for FormatStmtClassDef {
             }
         }
 
-        write!(
-            f,
-            [
-                text(":"),
-                trailing_comments(trailing_definition_comments),
-                block_indent(&body.format().with_options(SuiteKind::Class))
-            ]
-        )
+        // If the body contains only an ellipsis, format as:
+        // ```python
+        // class A: ...
+        // ```
+        if f.options().source_type().is_stub() && contains_only_an_ellipsis(body) {
+            write!(
+                f,
+                [
+                    text(":"),
+                    space(),
+                    &body.format().with_options(SuiteKind::Class),
+                    hard_line_break()
+                ]
+            )
+        } else {
+            write!(
+                f,
+                [
+                    text(":"),
+                    trailing_comments(trailing_definition_comments),
+                    block_indent(&body.format().with_options(SuiteKind::Class))
+                ]
+            )
+        }
```

Comment:
I think we need to generalize this logic because black doesn't just apply this to classes and functions but to any compound statement (it seems). 


```python
while True: ...

for i in []: ...

if True: ...

with True: ...

match x:
    case 1: ...
```


We can can take a similar approach as I took in https://github.com/astral-sh/ruff/pull/6593/files#diff-2a7940151585697a7844053afc3fc5c7beafc616a9e1dc49ec4d43b98258dc35 with `clause_header` 

```
pub(crate) struct FormatClauseBody<'a> {
	body: &'a Suite
	kind: SuiteKind,
	trailing_comments: &'a [SourceComment],
}

impl FormatClauseBody {
	#[must_use]
	pub(crate) fn with_kind(mut self, kind: SuiteKind)  -> Self {
		self.kind = kind;
		self
	}
}

pub(crate) fn clause_body(body: &Suite, trailing_colon_comments: &'a [SourceComment]) -> FormatClauseBody {
	FormatClauseBody { body, kind: SuiteKind::default(), trailing_comments }
}

impl Format<PyFormatContext<'_>> or FormatClauseBody<'_> {
  fn fmt(&self, f: &mut Formatter<PyFormatContext<'ast>>) -> FormatResult<()> {
		if f.options().source_type().is_stub() && contains_only_an_ellipsis(self.body, f.context().comments()) && self.trailing_comments.is_empty(){
			self.body.format().with_options(self.kind).fmt(f)
		} else {
			self.block_indent(&body.format().with_options(self.kind)).fmt(f)
		}
  }
}
```

And I think you have to extend `contains_only_an_ellipsis` to assert that the `...` has no comments by using `f.context().comments().has_comments(stmt)`

Reply:
Oh, it seems `has_comments` no longer exists on `Comments`. You can re-introduce it by adding the following method

```rust
#[inline]
    pub(crate) fn has_comments<T>(&self, node: T) -> bool
    where
        T: Into<AnyNodeRef<'a>>,
    {
        self.data
            .comments
            .has(&NodeRefEqualityKey::from_ref(node.into()))
    }
```